### PR TITLE
feat(channel): add `ForumChannel.default_layout`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

~~https://github.com/discord/discord-api-docs/pull/5693~~ https://github.com/discord/discord-api-docs/pull/5745

Does not seem to be supported on channel creation, and no audit log entry is being created when modifying the field.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/enums.py [hunk 1]
```diff
@@ -812,6 +813,12 @@ class ThreadSortOrder(Enum):
     creation_date = 1
 
 
+class ThreadLayout(Enum):
+    not_set = 0
+    list_view = 1
+    gallery_view = 2
```
Comment (Victorsitou) [inline disnake/enums.py:819]: It would be nice to have `list` and `gallery` aliases, in my opinion

Comment (shiftinv) [reply disnake/enums.py]: Aliases seem a bit much for such a fairly insignificant field; I've updated the enum in 6b64137322a593892a602ece1b5523629d6801fe to use those names instead, but haven't completely decided yet.
Enum members are usually named according to the api docs, so we may want to consider keeping the original ones. I'll keep this discussion open for now.

**Review by onerandomusername — CHANGES_REQUESTED** [review]
imo we should use the official documented names and not make aliases

File: disnake/enums.py (hunk 1 above)
Comment (Victorsitou) [reply disnake/enums.py:821]: I'm fine with using the API names, I don't think people will complain about it🤷 

You can resolve this :)

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**